source/Irrlicht/CBAWMeshFileLoader.cpp

```diff
@@ -208,10 +211,91 @@ void* CBAWMeshFileLoader::tryReadBlobOnStack(const SBlobData & _data, SContext &
 		dst = _stackPtr;
 	else
 		dst = malloc(_data.header->blobSizeDecompr);
+
+	const bool compressed = (_data.header->compressionType != core::Blob::EBCT_RAW);
```

Comment:
i dont think this is a valid way to check for compression (could have encryption instead of compression too)

Reply:
I know, changes will come along with encryption

Reply:
O, they already came actually